firedrake/interpolation.py

```diff
@@ -280,22 +327,39 @@ def _interpolator(V, tensor, expr, subset, arguments, access):
         parloop_args.append(tensor(access, V.cell_node_map()))
     else:
         assert access == op2.WRITE  # Other access descriptors not done for Matrices.
-        parloop_args.append(tensor(op2.WRITE, (V.cell_node_map(),
-                                               arguments[0].function_space().cell_node_map())))
+        if trans_mesh:
+            cmap = cell_parent_cell_fs_node_map
+        else:
+            cmap = arguments[0].function_space().cell_node_map()
+        parloop_args.append(tensor(op2.WRITE, (V.cell_node_map(), cmap)))
     if oriented:
-        co = mesh.cell_orientations()
+        co = target_mesh.cell_orientations()
         parloop_args.append(co.dat(op2.READ, co.cell_node_map()))
     if needs_cell_sizes:
-        cs = mesh.cell_sizes
+        cs = target_mesh.cell_sizes
         parloop_args.append(cs.dat(op2.READ, cs.cell_node_map()))
     for coefficient in coefficients:
-        m_ = coefficient.cell_node_map()
-        parloop_args.append(coefficient.dat(op2.READ, m_))
+        get_composed_map = (
+            trans_mesh
+            and (
+                isinstance(target_mesh.topology,
+                           firedrake.mesh.VertexOnlyMeshTopology)
+                and not isinstance(
+                    coefficient.function_space().mesh().topology,
+                    firedrake.mesh.VertexOnlyMeshTopology)
+            )
+        )
+        if get_composed_map:
+            m_ = compose_map_and_cache(target_mesh.cell_parent_cell_map, coefficient.cell_node_map())
+            parloop_args.append(coefficient.dat(op2.READ, m_))
```

Comment:
Note we don't need a composed map for the pyop2 output variable (`tensor`) since we want our result to be defined on the target mesh cells which we will loop over.